docker/Dockerfile

```diff
@@ -0,0 +1,17 @@
+# TOGO: gpg verify
+FROM openjdk:8
+ARG SCALAFMT_VERSION
+ARG SCALA_VERSION=2.12
+ARG SCALAFMT_INSTALL_LOCATION=/usr/local/bin/scalafmt
+RUN set -ex \
+    && apt-get update -y \
+    && rm -rf /var/lib/apt/lists/* \
+    && curl -sSLo coursier https://git.io/coursier-cli \
+    && chmod u+x coursier \
+    && ./coursier bootstrap "org.scalameta:scalafmt-cli_$SCALA_VERSION:$SCALAFMT_VERSION" \
```

Comment:
This assumes the CLI is already published to Maven Central. It can take 1-2hrs for the sync to complete so I wonder if want to build the binary instead. If the docker image is running on Linux we may as well use the native-image binary from here https://github.com/scalameta/scalafmt/blob/8f0ca529c02e4c9832d7bdc10e148bb50e2a2e3d/.github/workflows/ci.yml#L49

Maybe we can somehow depend on the native image artifact directly in a subsequent job by moving this logic to `ci.yml` 🤔 

Then we can also remove the dependency on the JDK, which could slim down the docker image.

Reply:
Okay, I see ☹️ .  Let me toy around with that.

Reply:
It's possible to download artifacts from other GH Action jobs using https://github.com/actions/download-artifact

Reply:
You should be able to test this in the PR by commenting out the `docker logic` and `docker push` commands. We link native images on every PR

Reply:
Ok, need to switch to some other work but will come back to this asap.

One question - can we ensure that `release` is run after `ci`?  Where is that ordering defined?

Reply:
You can't ensure that `release` runs after `ci` but you can move the release job into the `ci.yml` file to work around it. Here is an example workflow that does something similar https://github.com/mroth/scalafmt-native/blob/c0932c026050ea87da68337e2345f676f23414c4/.github/workflows/build.yml#L100